cmd/kubeadm/app/cmd/init.go

```diff
@@ -205,15 +210,118 @@ func NewInit(cfgPath string, cfg *kubeadmapi.MasterConfiguration, skipPreFlight
 		fmt.Println("\t(/etc/systemd/system/kubelet.service.d/10-kubeadm.conf should be edited for this purpose)")
 	}
 
-	return &Init{cfg: cfg}, nil
+	var deploymentTypeValid bool
+	for _, supportedDT := range deploymentTypes {
```

Comment:
@luxas since deployment type is a string, I can use `k8s.io/kubernetes/pkg/util/sets` here, right?